src/coprocessor/statistics/analyze.rs

```diff
@@ -220,7 +228,31 @@ impl<S: Snapshot> SampleBuilder<S> {
                     pk_builder.append(&v);
                 }
             }
-            for (collector, val) in collectors.iter_mut().zip(cols_iter) {
+            for (i, (collector, val)) in collectors.iter_mut().zip(cols_iter).enumerate() {
+                if self.cols_info[i].get_tp() == FieldTypeTp::TinyBlob as i32
+                    || self.cols_info[i].get_tp() == FieldTypeTp::MediumBlob as i32
+                    || self.cols_info[i].get_tp() == FieldTypeTp::Blob as i32
+                    || self.cols_info[i].get_tp() == FieldTypeTp::VarChar as i32
+                    || self.cols_info[i].get_tp() == FieldTypeTp::VarString as i32
+                    || self.cols_info[i].get_tp() == FieldTypeTp::String as i32
+                {
+                    let sorted_val = match_template_collator! {
+                        TT, match Collation::from_i32(self.cols_info[i].get_collation())? {
+                            Collation::TT => {
+                                let mut mut_val = val.as_slice();
+                                let decoded_val = table::decode_col_value(&mut mut_val, &mut EvalContext::default(), &self.cols_info[i])?;
```

Comment:
prefer not to use functions in `table::`

@iosmanthus Please give some advice for better alternatives